src/CHIPBindings.cc

```diff
@@ -2439,25 +2438,31 @@ hipError_t hipHostRegister(void *HostPtr, size_t SizeBytes,
                            unsigned int Flags) {
   CHIP_TRY
   CHIPInitialize();
-  NULLCHECK(HostPtr);
+  if (!HostPtr || !SizeBytes)
+    RETURN(hipErrorInvalidValue);
 
   // TODO fixOpenCLTests - make this a class
-  if (Flags)
-    switch (Flags) {
-    case hipHostRegisterDefault:
-      break;
-    case hipHostRegisterMapped:
-      break;
-    case hipHostRegisterPortable:
-      break;
-    default:
-      CHIPERR_LOG_AND_THROW("Invalid hipHostRegister flag passed",
-                            hipErrorInvalidValue);
-    }
+  if (Flags) {
+    // Currently, the flags are ignored. This only exists to satisfy hip-tests.
+
+    // First 4 bits are valid flag bits. This includes flags from CUDA which are
+    // not supported or documented in HIP.
+    constexpr unsigned FlagMask = (1u << 4u) - 1u;
+
+    if (Flags & ~FlagMask) // Has invalid flags
+      CHIPERR_LOG_AND_THROW("Invalid hipHostRegister flags passed",
+                             hipErrorInvalidValue);
+
+    if (Flags & hipHostRegisterIoMemory)
+      CHIPERR_LOG_AND_THROW("Unsupported hipHostRegisterIoMemory flag",
+                             hipErrorInvalidValue);
+  }
 
   void *DevPtr;
-  auto Err = hipMalloc(&DevPtr, SizeBytes);
-  ERROR_IF(Err != hipSuccess, Err);
+  if (hipMalloc(&DevPtr, SizeBytes) != hipSuccess)
+    // Translate hipOutOfMemory to hipErrorInvalidValue. The latter is
+    // the one hip-tests suite expects in case of OoM.
+    RETURN(hipErrorInvalidValue);
```

Comment:
What does the spec say about this? We shouldn't rely on tests as ground-truth

Reply:
Acknowledged, chose to follow hip-tests expectation until it reflects the API.